test/call_DX_methods.js

```diff
@@ -0,0 +1,445 @@
+// Testing Generic Scheme:
+
+// // 0
+// // before
+// make sure accounts have rep (GEN)
+
+// // 1
+// // get call data for
+// DX.updateThresholdNewTokenPair(uint _thresholdNewTokenPair)
+// DX.updateThresholdNewAuction(uint _thresholdNewAuction)
+
+// DX.startMasterCopyCountdown(address _masterCopy)
+// //timeskip 30 days
+// DX.updateMasterCopy()
+
+// DX.updateApprovalOfToken(address[] token, bool approved)
+
+// DX.initiateEthUsdOracleUpdate(PriceOracleInterface _ethUSDOracle)
+// //timeskip 30 days
+// DX.updateEthUSDOracle()
+
+// DX.updateAuctioneer(address _auctioneer)
+
+// // basically every method with onlyAuctioneer modifier
+
+// // 2
+// call GenericScheme.proposeCall(Avatar _avatar, bytes _callData) -> proposalId
+
+// which calls GenesisProtocol.propose(uint _numOfChoices, bytes32 _paramsHash,address _proposer,address _organization)
+
+// GenericProtocol.proposals(proposalId) -> current proposal
+
+// // 3
+// // voting
+// GenericProtocol.vote(bytes32 _proposalId, {from: accountWithRep})
+// winning vote calls GenericProtocol._execute(bytes32 _proposalId)
+
+/* global artifacts, web3, contract, it, before, beforeEach, after, afterEach, assert, expect, should */
+
+const MgnToken = artifacts.require('TokenFRT')
+const DxLockMgnForRep = artifacts.require('DxLockMgnForRep')
+const DxReputation = artifacts.require('DxReputation')
+const DxAvatar = artifacts.require('DxAvatar')
+const GenericScheme = artifacts.require('GenericScheme')
+const GenesisProtocol = artifacts.require('GenesisProtocol')
+const DutchExchange = artifacts.require('DutchExchange')
+const DutchExchangeProxy = artifacts.require('DutchExchangeProxy')
+const BN = require('bn.js')
+const {
+  increaseTimeAndMine,
+  takeSnapshot,
+  revertSnapshot,
+  getTimestamp
+} = require('../src/helpers/web3helpers')(web3)
+
+const LOCK_AMOUNT = 2001
+const YES_VOTE = new BN(1)
+const NO_VOTE = new BN(2)
+const EXECUTED_STATE = new BN(2)
+
+// won't work because we transferred ownership to DxController
+// only schemes registered with it can mint tokens
+// async function mintReputation(account, amount, dxReputation) {
+//   if (amount) {
+//     const txResult = await dxReputation.mint(account, amount)
+//     console.log(`REP minted:
+// - Transaction: ${txResult.tx}
+// - Gas used: ${txResult.receipt.gasUsed}
+// `)
+//   }
+// }
+const callForAll = (accounts, cb) => { return Promise.all(accounts.map(cb)) }
+
+contract('Execute updateAuctioneer proposal', (accounts) => {
```

Comment:
where does the GenericS parameters set is perform ?

Reply:
can you please add a call to validate the contractToCall parameter ?

Reply:
Some parameters were indeed incorrectly set. And auctioneer was different from the expected one.
Thank you for pointing me in this direction.

Tests now work, proposals execute and final values are what is expected, but there's still no `ProposalDeleted and ProposalExecuted` events. Do you have any idea why?

Reply:
Parameters are set in migration 13